src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -61,21 +55,18 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             if (padding == null)
                 throw new ArgumentNullException(nameof(padding));
 
-            if (padding == RSAEncryptionPadding.Pkcs1)
-            {
-                return Decrypt(data, fOAEP: false);
-            }
-            else if (padding == RSAEncryptionPadding.OaepSHA1)
-            {
-                // For compat, this prevents OaepSHA2 options as fOAEP==true will cause Decrypt to use OaepSHA1
-                return Decrypt(data, fOAEP: true);
-            }
-            else
-            {
+            return
+                padding == RSAEncryptionPadding.Pkcs1 ? Decrypt(data, fOAEP: false) :
+                padding == RSAEncryptionPadding.OaepSHA1 ? Decrypt(data, fOAEP: true) : // For compat, this prevents OaepSHA2 options as fOAEP==true will cause Decrypt to use OaepSHA1
                 throw PaddingModeNotSupported();
-            }
         }
 
+        public override bool TryDecrypt(ReadOnlySpan<byte> source, Span<byte> destination, RSAEncryptionPadding padding, out int bytesWritten) =>
+            padding == null ? throw new ArgumentNullException(nameof(padding)) :
+            source.Length > (KeySize / 8) ? throw new CryptographicException(SR.Format(SR.Cryptography_Padding_DecDataTooBig, Convert.ToString(KeySize / 8))) :
+            padding == RSAEncryptionPadding.Pkcs1 || padding == RSAEncryptionPadding.OaepSHA1 ? _impl.TryDecrypt(source, destination, padding, out bytesWritten) :
+            throw PaddingModeNotSupported();
```

Comment:
I actually find this easier, as it's a "if/then, if/then, if/then, ..." in a concise format.  I can change it if you really don't like it, though.